src/utils/theme-manager.ts

```diff
@@ -0,0 +1,59 @@
+import { storage } from '#imports'
+import { THEME_MODE_KEY } from './constants/storage-keys'
+
+export type ThemeMode = 'light' | 'dark' | 'system'
+
+let cachedThemeMode: ThemeMode = 'system'
```

Comment:
shouldn't have theme `cachedThemeMode` and `isInitialized`

because the service worker (background script) will be inactivate after certain time and them these variable will all be unavailable.

You should store all persistent things in storage API, and use Provider to get and set value into storage API.